tools/topology/topology2/development/cavs-nocodec-asrc.conf

```diff
@@ -0,0 +1,758 @@
+			id	0
+			bclk_freq	3072000
+			tdm_slot_width	32
+			# TODO: remove this. Needs alsaptlg change.
+			Object.Base.link_config.1 {
+				clock_source	1
+			}
+		}
+	}
+]
+
+Object.Dai.DMIC [
+	{
+		dai_index 0
+		name			$DMIC0_NAME
+		id 			$DMIC0_ID
+		driver_version		$DMIC_DRIVER_VERSION
+		io_clk			$DMIC_IO_CLK
+		clk_min		500000
+		clk_max		4800000
+		unmute_ramp_time_ms	200
+		# num_pdm_active should always set to 2 but depending on the number of DMIC's
+		# the mic's are enabled or disabled in each PDM.
+		num_pdm_active 	2
+
+		Object.Base.hw_config.1 {
+			id	0
+			name	"DMIC0"
+		}
+
+		# PDM controller config
+		Object.Base.pdm_config.1 {
+			mic_a_enable	$PDM0_MIC_A_ENABLE
+			mic_b_enable	$PDM0_MIC_B_ENABLE
+			ctrl_id	0
+		}
+
+		Object.Base.pdm_config.2 {
+			ctrl_id	1
+			mic_a_enable	$PDM1_MIC_A_ENABLE
+			mic_b_enable	$PDM1_MIC_B_ENABLE
+		}
+	}
+]
+
+# Dmic-->ASRC-->host copier PCM ASRC Capture3
+# SSP0-->host copier PCM Port0
+# SSP1-->ASRC-->host copier PCM Port1
+# SSP2-->ASRC-->host copier PCM Port2
+Object.Pipeline.dai-copier-asrc-capture [
+	{
+		index		1
+		core_id		$DMIC_CORE_ID
+		Object.Widget.dai-copier.1 {
+			dai_index	0
+			dai_type	"DMIC"
+			copier_type	"DMIC"
+			type		dai_out
+			stream_name	$DMIC0_NAME
+			node_type $DMIC_LINK_INPUT_CLASS
+			num_input_audio_formats	2
+			Object.Base.input_audio_format [
+				{
+					in_bit_depth		32
+					in_valid_bit_depth	32
+				}
+				{
+					in_channels		4
+					in_bit_depth		32
+					in_valid_bit_depth	32
+					in_ch_cfg	$CHANNEL_CONFIG_3_POINT_1
+					in_ch_map	$CHANNEL_MAP_3_POINT_1
+				}
+			]
+			num_output_audio_formats	2
+			Object.Base.output_audio_format [
+				{
+					out_bit_depth		32
+					out_valid_bit_depth	32
+				}
+				{
+					out_channels		4
+					out_bit_depth		32
+					out_valid_bit_depth	32
+					out_ch_cfg	$CHANNEL_CONFIG_3_POINT_1
+					out_ch_map	$CHANNEL_MAP_3_POINT_1
+				}
+			]
+		}
+
+		Object.Widget.asrc."1" {
+			rate_in 48000
+			rate_out 48000
+			asynchronous_mode	1
+			operation_mode		1
+			num_input_audio_formats	2
+			Object.Base.input_audio_format [
+				{
+					in_bit_depth		32
+					in_valid_bit_depth	32
+				}
+				{
+					in_channels		4
+					in_bit_depth		32
+					in_valid_bit_depth	32
+					in_ch_cfg	$CHANNEL_CONFIG_3_POINT_1
+					in_ch_map	$CHANNEL_MAP_3_POINT_1
+				}
+			]
+			num_output_audio_formats	2
+			Object.Base.output_audio_format [
+				{
+					out_bit_depth		32
+					out_valid_bit_depth	32
+				}
+				{
+					out_channels		4
+					out_bit_depth		32
+					out_valid_bit_depth	32
+					out_ch_cfg	$CHANNEL_CONFIG_3_POINT_1
+					out_ch_map	$CHANNEL_MAP_3_POINT_1
+				}
+			]
+		}
+
+		Object.Widget.pipeline."1" {
+			core $DMIC_CORE_ID
+		}
+	}
+
+	{
+		index		7
+		Object.Widget.dai-copier."1" {
+			dai_index	1
+			dai_type	"SSP"
+			type		dai_out
+			copier_type	"SSP"
+			stream_name	"NoCodec-1"
+			node_type	$I2S_LINK_INPUT_CLASS
+		}
+
+		Object.Widget.asrc."1" {
+			rate_in 48000
+			rate_out 44100
```

Comment:
Is this a capture ASRC with 44.1 kHz rate out to PCM?

Reply:
No, it is 48kHz